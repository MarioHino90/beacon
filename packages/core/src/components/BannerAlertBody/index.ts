export { BannerAlertBody } from "./BannerAlertBody";

export type { BannerAlertBodyProps } from "./types";
