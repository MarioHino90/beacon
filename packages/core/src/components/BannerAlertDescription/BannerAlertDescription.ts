import styled from "styled-components";
import classNames from "classnames";
import { getColor, getFontSize, getFontWeight } from "@braze/beacon-styling";

import { BannerAlertDescriptionProps } from "./types";

const getAttrs = ({
  className,
}: BannerAlertDescriptionProps): BannerAlertDescriptionProps => ({
  className: classNames("bcl-banner-alert-description", className),
});

export const BannerAlertDescription = styled.div.attrs(
  getAttrs
)<BannerAlertDescriptionProps>`
  font-size: ${getFontSize("body", "xs")};
  font-weight: ${getFontWeight("regular")};
  color: ${getColor("gray", 900)};
`;

BannerAlertDescription.displayName = "BannerAlertDescription";
