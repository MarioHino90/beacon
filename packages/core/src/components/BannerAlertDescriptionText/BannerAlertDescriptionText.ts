import styled from "styled-components";
import classNames from "classnames";

import { BannerAlertDescriptionTextProps } from "./types";

const getAttrs = ({
  className,
}: BannerAlertDescriptionTextProps): BannerAlertDescriptionTextProps => ({
  className: classNames("bcl-banner-alert-description-text", className),
});

export const BannerAlertDescriptionText = styled.span.attrs(
  getAttrs
)<BannerAlertDescriptionTextProps>`
  margin: 0 10px 0 0;
`;

BannerAlertDescriptionText.displayName = "BannerAlertDescriptionText";
