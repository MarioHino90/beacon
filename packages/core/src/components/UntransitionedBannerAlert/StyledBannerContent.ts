import styled from "styled-components";

export const StyledBannerContent = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;
