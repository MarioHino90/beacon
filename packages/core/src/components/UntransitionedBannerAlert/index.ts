export { UntransitionedBannerAlert } from "./UntransitionedBannerAlert";

export type { UntransitionedBannerAlertProps } from "./types";
