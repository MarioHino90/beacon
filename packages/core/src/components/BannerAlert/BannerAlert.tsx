import React, { FC } from "react";
import classNames from "classnames";

import { UntransitionedBannerAlert } from "../UntransitionedBannerAlert";

import { BannerAlertProps } from "./types";
import { StyledBannerTransition } from "./StyledBannerTransition";

export const BannerAlert: FC<BannerAlertProps> = ({
  isOpen,
  className,
  intent,
  children,
  ...props
}) => {
  return (
    <StyledBannerTransition
      className={classNames(["bcl-banner-alert-transition", className])}
      isOpen={isOpen}
      {...props}
    >
      <UntransitionedBannerAlert className={className} intent={intent}>
        {children}
      </UntransitionedBannerAlert>
    </StyledBannerTransition>
  );
};

BannerAlert.displayName = "BannerAlert";
