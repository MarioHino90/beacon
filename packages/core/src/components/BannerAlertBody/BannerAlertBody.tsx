import React, { useContext } from "react";

import { BannerAlertClose } from "../BannerAlertClose";
import { BannerAlertIcon } from "../BannerAlertIcon";
import { IntentContext } from "../BannerAlert/IntentContext";

import { BannerAlertBodyProps } from "./types";
import { StyledBannerBodyWrapper } from "./StyledBannerBodyWrapper";
import { StyledBannerBody } from "./StyledBannerBody";
import { StyledBodyWrapper } from "./StyledBodyWrapper";

export const BannerAlertBody = ({
  children,
  onClose,
  Icon,
  ...props
}: BannerAlertBodyProps): React.ReactElement => {
  const intent = useContext(IntentContext);

  return (
    <StyledBannerBodyWrapper
      className="bcl-banner-alert-body"
      hasClose={!!onClose}
      {...props}
    >
      <StyledBannerBody className="bcl-banner-alert-inner-body">
        <div className="bcl-banner-alert-body-icon">
          {Icon || <BannerAlertIcon intent={intent} />}
        </div>
        <StyledBodyWrapper className="bcl-banner-alert-body-children-wrapper">
          {children}
        </StyledBodyWrapper>
      </StyledBannerBody>
      {onClose ? <BannerAlertClose intent={intent} onClose={onClose} /> : null}
    </StyledBannerBodyWrapper>
  );
};

BannerAlertBody.displayName = "BannerAlertBody";
