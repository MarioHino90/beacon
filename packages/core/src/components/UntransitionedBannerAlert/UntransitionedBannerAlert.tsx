import React from "react";
import classNames from "classnames";

import { SimpleBannerAlert } from "../SimpleBannerAlert";
import { IntentContextProvider } from "../BannerAlert/IntentContext";

import { UntransitionedBannerAlertProps } from "./types";
import { StyledBannerContent } from "./StyledBannerContent";

export const UntransitionedBannerAlert = ({
  intent,
  children,
  className,
}: UntransitionedBannerAlertProps): React.ReactElement => (
  <IntentContextProvider intent={intent}>
    <SimpleBannerAlert
      className={classNames("bcl-banner-alert-untransitioned", className)}
    >
      <StyledBannerContent>{children}</StyledBannerContent>
    </SimpleBannerAlert>
  </IntentContextProvider>
);

UntransitionedBannerAlert.displayName = "UntransitionedBannerAlert";
