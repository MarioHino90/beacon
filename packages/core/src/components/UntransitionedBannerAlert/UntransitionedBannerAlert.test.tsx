import React from "react";
import { shallow } from "enzyme";

import { Intent } from "../BannerAlert";
import { BannerAlertLink } from "../BannerAlertLink";
import { BannerAlertBody } from "../BannerAlertBody";
import { BannerAlertTitle } from "../BannerAlertTitle";
import { BannerAlertDescription } from "../BannerAlertDescription";
import { BannerAlertDescriptionText } from "../BannerAlertDescriptionText";

import { UntransitionedBannerAlert } from "./UntransitionedBannerAlert";

describe("UntransitionedBannerAlert", () => {
  const intents: Intent[] = ["success", "danger", "warning", "info"];

  it.each(intents)("should render correctly for intent %s", (intent) => {
    const wrapper = shallow(
      <UntransitionedBannerAlert intent={intent}>
        <BannerAlertBody>
          <BannerAlertTitle>Lorem ipsum dolor sit amet</BannerAlertTitle>
        </BannerAlertBody>
      </UntransitionedBannerAlert>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if passed link info", () => {
    const wrapper = shallow(
      <UntransitionedBannerAlert intent="success">
        <BannerAlertBody>
          <BannerAlertTitle>Lorem ipsum dolor sit amet</BannerAlertTitle>
          <BannerAlertLink onClick={jest.fn().mockName("onClick")}>
            Click Me!
          </BannerAlertLink>
        </BannerAlertBody>
      </UntransitionedBannerAlert>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if passed a description", () => {
    const wrapper = shallow(
      <UntransitionedBannerAlert intent="success">
        <BannerAlertBody>
          <BannerAlertTitle>Lorem ipsum dolor sit amet</BannerAlertTitle>
          <BannerAlertDescription>
            <BannerAlertDescriptionText>
              Some more information about the banner...
            </BannerAlertDescriptionText>
          </BannerAlertDescription>
        </BannerAlertBody>
      </UntransitionedBannerAlert>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should render correctly if passed a className", () => {
    const wrapper = shallow(
      <UntransitionedBannerAlert intent="success" className="new-banner">
        <BannerAlertBody>
          <BannerAlertTitle>Some banner information here...</BannerAlertTitle>
        </BannerAlertBody>
      </UntransitionedBannerAlert>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
