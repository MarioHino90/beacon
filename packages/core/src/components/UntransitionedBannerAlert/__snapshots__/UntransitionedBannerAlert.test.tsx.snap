// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UntransitionedBannerAlert should render correctly for intent danger 1`] = `
<IntentContextProvider
  intent="danger"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly for intent info 1`] = `
<IntentContextProvider
  intent="info"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly for intent success 1`] = `
<IntentContextProvider
  intent="success"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly for intent warning 1`] = `
<IntentContextProvider
  intent="warning"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly if passed a className 1`] = `
<IntentContextProvider
  intent="success"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned new-banner"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Some banner information here...
        </BannerAlertTitle>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly if passed a description 1`] = `
<IntentContextProvider
  intent="success"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
        <BannerAlertDescription>
          <BannerAlertDescriptionText>
            Some more information about the banner...
          </BannerAlertDescriptionText>
        </BannerAlertDescription>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;

exports[`UntransitionedBannerAlert should render correctly if passed link info 1`] = `
<IntentContextProvider
  intent="success"
>
  <SimpleBannerAlert
    className="bcl-banner-alert-untransitioned"
  >
    <StyledBannerContent>
      <BannerAlertBody>
        <BannerAlertTitle>
          Lorem ipsum dolor sit amet
        </BannerAlertTitle>
        <BannerAlertLink
          onClick={[MockFunction onClick]}
        >
          Click Me!
        </BannerAlertLink>
      </BannerAlertBody>
    </StyledBannerContent>
  </SimpleBannerAlert>
</IntentContextProvider>
`;
